lib/Graph/Grad.cpp

```diff
@@ -111,7 +112,7 @@ Function *glow::differentiate(Function *F, const TrainingConfig &conf,
 
     if (N->getKind() == Kind::SaveNodeKind) {
       // Swap the src and dest. Send the Zero value as gradient for both sides.
-      auto *X = new SplatNode(N->getName(),
+      auto *X = new SplatNode(N->getName().str() + "_grad",
```

Comment:
Since this pattern: `x.str() + "_grad"` is used many times in your PR, I'd suggest creating a helper function for it. Something like `getGradName`.